packages/contracts-bedrock/test/L1/SystemConfigInterop.t.sol

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Testing utilities
+import { CommonTest } from "test/setup/CommonTest.sol";
+
+// Libraries
+import { Constants } from "src/libraries/Constants.sol";
+import { StaticConfig } from "src/libraries/StaticConfig.sol";
+import { GasPayingToken } from "src/libraries/GasPayingToken.sol";
+
+// Target contract dependencies
+import { SystemConfig } from "src/L1/SystemConfig.sol";
+import { SystemConfigInterop } from "src/L1/SystemConfigInterop.sol";
+import { OptimismPortalInterop } from "src/L1/OptimismPortalInterop.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { ConfigType } from "src/L2/L1BlockInterop.sol";
+
+contract SystemConfigInterop_Test is CommonTest {
+    /// @notice Marked virtual to be overridden in
+    ///         test/kontrol/deployment/DeploymentSummary.t.sol
+    function setUp() public virtual override {
+        super.enableInterop();
+        super.setUp();
+    }
+
+    /// @dev Tests that the gas paying token can be set.
+    function testFuzz_setGasPayingToken_succeeds(
+        address _token,
+        string calldata _name,
+        string calldata _symbol
+    )
+        public
+    {
+        vm.assume(_token != address(0));
+        vm.assume(_token != Constants.ETHER);
+        // don't use vm's address
+        vm.assume(_token != address(vm));
+        // don't use console's address
+        vm.assume(_token != CONSOLE);
+        // don't use create2 deployer's address
+        vm.assume(_token != CREATE2_FACTORY);
+        // don't use default test's address
+        vm.assume(_token != DEFAULT_TEST_CONTRACT);
+        // don't use multicall3's address
+        vm.assume(_token != MULTICALL3_ADDRESS);
+
+        vm.assume(bytes(_name).length <= 32);
+        vm.assume(bytes(_symbol).length <= 32);
+
+        vm.mockCall(_token, abi.encodeWithSelector(ERC20.decimals.selector), abi.encode(18));
+        vm.mockCall(_token, abi.encodeWithSelector(ERC20.name.selector), abi.encode(_name));
+        vm.mockCall(_token, abi.encodeWithSelector(ERC20.symbol.selector), abi.encode(_symbol));
+
+        vm.expectCall(
+            address(optimismPortal),
+            abi.encodeCall(
+                OptimismPortalInterop.setConfig,
+                (
+                    ConfigType.GAS_PAYING_TOKEN,
+                    StaticConfig.encodeSetGasPayingToken({
+                        _token: _token,
+                        _decimals: 18,
+                        _name: GasPayingToken.sanitize(_name),
+                        _symbol: GasPayingToken.sanitize(_symbol)
+                    })
+                )
+            )
+        );
+
+        _cleanStorageAndInit(_token);
```

Comment:
had to create an additional internal function (`_cleanStorageAndInit`) since otherwise we were hitting stack too deep in `testFuzz_setGasPayingToken_succeeds`